packages/shared/src/database/migrations/1662077882511-DataLayerRework.ts

```diff
@@ -0,0 +1,131 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class DataLayerRework1662077882511 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(`
+      DROP TABLE "web_page_package";
+    `);
+
+    await queryRunner.query(`
+      DROP TABLE "web_page";
+    `);
+
+    await queryRunner.query(`
+      CREATE TABLE "hostname" (
+        "id" serial primary key,
+        "hostname" text not null
+      );
+      
+      CREATE UNIQUE INDEX "hostname_hostname" ON "hostname" ("hostname");
+    `);
+
+    await queryRunner.query(`
+      CREATE TABLE "web_page" (
+        "id" serial primary key,
+        "hostname_id" integer not null references "hostname"("id"),
+        "path" text not null,
+        "created_at" timestamp not null default now()
+      );
+      
+      CREATE UNIQUE INDEX "web_page_hostname_id_path" ON "web_page" ("hostname_id", "path");
+    `);
+
+    await queryRunner.query(`
+      CREATE TABLE "web_page_scan" (
+        "id" serial primary key,
+        "web_page_id" integer not null references "web_page"("id"),
+        "status" text not null,
+        "scan_result" jsonb default null,
+        "created_at" timestamp not null default now(),
+        "finished_at" timestamp
+      );
+      
+      CREATE INDEX "web_page_scan_web_page_id_created_at" ON "web_page_scan" ("web_page_id", "created_at" DESC);
+    `);
+
+    await queryRunner.query(`
+      CREATE TABLE "package_usage_by_hostname_projection" (
+        "id" serial primary key,
+        "hostname_id" integer not null references "hostname"("id"),
+        "source_scan_id" integer not null references "web_page_scan"("id"),
```

Comment:
I'd recommend against using the explicit `references` clause, as it may cause problems more than it solves. I believe that in most cases conventional references are enough (e.g. on the application level).

Reply:
Could you please elaborate? I've never encountered problems with FKs, they are integral part of any relational DB.